File: python/iniciante/problem1040.py
# Inputs

a, b, c, d = input().split()
a = float(a)
b = float(b)
c = float(c)
d = float(d)

# Calculations and Outputs

mean = (2*a + 3*b + 4*c + 1*d)/(2 + 3 + 4 + 1)

print("Media: %.1f" % mean)
if mean >= 7.0:
    print("Aluno aprovado.")
elif mean < 5.0:
    print("Aluno reprovado.")
else:
    print("Aluno em exame.")
    e = float(input())
    print("Nota do exame: %.1f" % e)
    final = (mean + e)/2
    if final >= 5.0:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    print("Media final: %.1f" % final)

File: go/01-iniciante/problem1040.go
package main

import "fmt"

func main() {
    var a, b, c, d float64
    fmt.Scanf("%f %f %f %f", &a, &b, &c, &d)

    mean := (a*2.0 + b*3.0 + c*4.0 + d*1.0)/10.0
    output := fmt.Sprintf("%.1f", mean)

    fmt.Println("Media:", output)

    if mean >= 7.0 {
        fmt.Println("Aluno aprovado.")
    } else if mean < 5.0 {
        fmt.Println("Aluno reprovado.")
    } else {
        fmt.Println("Aluno em exame.")

        var e float64
        fmt.Scanf("%f", &e)

        output = fmt.Sprintf("%.1f", e)

        fmt.Println("Nota do exame:", output)

        mean = (mean + e)/2.0
        output = fmt.Sprintf("%.1f", mean)

        if mean >= 5.0 {
            fmt.Println("Aluno aprovado.")
        } else {
            fmt.Println("Aluno reprovado.")
        }

        fmt.Println("Media final:", output)
    }
}
